File: apzkr-pzpi-21-11-milennyi-ivan/Task3-WebClient/src/models/Shepherd.ts

// Клас, що представляє пастуха
export class Shepherd {
    id: number | null; // Ідентифікатор пастуха
    username: string; // Ім'я користувача пастуха
    password: string; // Пароль пастуха
    name: string; // Ім'я пастуха
    surname: string; // Прізвище пастуха

    constructor(id: number | null, username: string, password: string, name: string, surname: string) {
        this.id = id;
        this.username = username;
        this.password = password;
        this.name = name;
        this.surname = surname;
    }
}

File: apzkr-pzpi-21-11-milennyi-ivan/Task1-Server/src/models/shepherd.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use utoipa::ToSchema;
use validator::Validate;

// Структура, що представляє пастуха
#[derive(Debug, Validate, Deserialize, Serialize, FromRow, ToSchema)]
pub(crate) struct Shepherd {
    id: Option<u64>, // Ідентифікатор пастуха
    #[validate(length(min = 1))]
    username: String, // Логін пастуха
    #[validate(length(min = 1))]
    password: String, // Пароль пастуха
    #[validate(length(min = 1))]
    name: String, // Ім'я пастуха
    #[validate(length(min = 1))]
    surname: String, // Прізвище пастуха
}

impl Shepherd {
    // Отримання ідентифікатора пастуха
    pub fn id(&self) -> Option<u64> {
        self.id
    }

    // Встановлення ідентифікатора пастуха
    pub fn set_id(&mut self, id: u64) -> () {
        self.id = Some(id);
    }

    // Отримання логіна пастуха
    pub fn username(&self) -> &str {
        &self.username
    }

    // Отримання пароля пастуха
    pub fn password(&self) -> &str {
        &self.password
    }

    // Отримання імені пастуха
    pub fn name(&self) -> &str {
        &self.name
    }

    // Отримання прізвища пастуха
    pub fn surname(&self) -> &str {
        &self.surname
    }
}
